Thinktecture.Relay.OnPremiseConnector/SignalR/RelayServerConnection.cs

```diff
@@ -126,10 +125,10 @@ public void RegisterOnPremiseTarget(string key, Uri baseUri)
                 await Start();
                 _logger.Info("Connected to relay server. #" + _id);
             }
-            catch (Exception ex)
+            catch (Exception)
             {
                 _logger.Info("***** ERROR WHILE CONNECTING: #" + _id);
-                Delay(5000).ContinueWith(_ => Start());
+                await Delay(5000).ContinueWith(_ => Start());
```

Comment:
@ManuelRauber : I don't know in deep the code of `Thinktecture.Relay.Server`. I did the suggested changes with adding `await` just following the suggestion of ReSharper, which seems to me very logical:
![delay](https://cloud.githubusercontent.com/assets/249117/14615282/2cd6d378-05a5-11e6-9010-a06993986a83.png)
I searched a little more in the Internet for close problems and could find for example a suggestion to call `.Unwrap` after `.ContinueWith`(see [this answer](http://stackoverflow.com/a/28587027/315935)), but it seems to me personally not needed. If I correctly interpret the last sentence of [the answer](http://stackoverflow.com/a/18965299/315935) and the above description (see "awaiting a task will "unwrap" aggregate exceptions") then the usage of additional `.Unwrap` call is not needed.

In any way it's very interesting, but complex problem. I think that complex understanding code is the problem of errors and side effects. Thus I looked once more what one really try to implement in the code fragment. The behavior looks like
```cs
await Task.Delay(5000);
await Start();
```
seems to me more readable. Above one uses the standard `Task.Delay`, which you use already in `TraceControllerTest` the form `await Task.Delay(1);`. I personally like the code written with `async` ... `await`, because the code is very easy to read and to understand.

On the other side the code still have some small problem. We start `Start()` once more time inside of `catch` block (in case of some execution inside of the first `Start()`). What behavior one want to implement? Is it some kind of protection against **only one exception** in the *first* execution of `Start()`? I'm not sure, which behavior one want to have. Probably one can place the `try catch` block inside of some infinitive loop (or in a loop with some cofigurable number of retry attempts) like the following:
```cs
var notFinished = true;
while (notFinished)
{
    try
    {
        await Start();
        _logger.Info("Connected to relay server. #" + _id);
        notFinished = false;
    }
    catch (Exception)
    {
        _logger.Info("***** ERROR WHILE CONNECTING: #" + _id);
        await Task.Delay(5000);
    }
}
```
I have to repeat that I'm not deep enough in the context of the code. I think that you can better decide yourself what to do here. I just recommend you to search in the whole code of the Solution for `Delay` and `Thread.Sleep` (`Thread.Sleep(1000);`, `Thread.Sleep(randomWaitTime);`) and to think about the usage of `await Task.Delay(someValue)` instead (like `await Task.Delay(1000);` or `await Task.Delay(randomWaitTime);`).

Best regards